rhp/v4/rhp.go

```diff
@@ -680,14 +684,15 @@ func NewContract(p HostPrices, cp RPCFormContractParams, hostKey types.PublicKey
 }
 
 // ContractCost calculates the cost to the host and renter for forming a contract.
-func ContractCost(cs consensus.State, p HostPrices, fc types.V2FileContract, minerFee types.Currency) (renter, host types.Currency) {
-	renter = fc.RenterOutput.Value.Add(p.ContractPrice).Add(minerFee).Add(cs.V2FileContractTax(fc))
+func ContractCost(cs consensus.State, fc types.V2FileContract, minerFee types.Currency) (renter, host types.Currency) {
+	contractCost := fc.HostOutput.Value.Sub(fc.TotalCollateral) // (contract price + locked collateral) - locked collateral
```

Comment:
This makes me nervous. I feel like it should be `.Sub(fc.MissedHostValue)` but I guess when forming a contract `MissedHostValue` and `TotalCollateral` are equal.